### PR TITLE
CLDC-3465 Allow clearing all error questions

### PR DESCRIPTION
**Pull request by kosiakkatrina**
This builds on top of https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data/pull/2472
and adds functionality to clear all button/link

### CODE REVIEW

File: app/controllers/check_errors_controller.rb [hunk 1]
```diff
@@ -8,9 +8,21 @@ def confirm_clear_answer
...
-    question_id = @related_question_ids.find { |id| !params[id].nil? }
-    @question = @log.form.get_question(question_id, @log)
     @page = @log.form.get_page(params[@log.model_name.param_key]["page_id"])
+
+    if params["clear_all"]
+      @questions_to_clear = @related_question_ids.map { |id|
+        question = @log.form.get_question(id, @log)
+        next if question.subsection.id == "setup"
+
+        question.page.questions.map(&:id)
```
Comment (RachaelBooth) [inline app/controllers/check_errors_controller.rb:18]: Should this be just question? I know we don't put more than one on a page very often (and only if they are linked), but I'd expect others on the same page to be related in their own right if they needed to be cleared. (Maybe that expectation is wrong though)

File: app/views/check_errors/confirm_clear_all_answers.html.erb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<% content_for :before_content do %>
+  <% content_for :title, "Are you sure you want to clear all?" %>
+<% end %>
+
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds-from-desktop">
+    <h1 class="govuk-heading-xl">
+      <%= content_for(:title) %>
+    </h1>
+    <p class="govuk-body">You've selected <%= @questions_to_clear.count %> answers to clear</p>
```
Comment (RachaelBooth) [inline app/views/check_errors/confirm_clear_all_answers.html.erb:10]: Do we need to have anything that takes into account anything that will be cleared as a result of clearing these? (That sounds really hard to identify, so maybe just a warning?)

File: app/controllers/check_errors_controller.rb (hunk 1 above)
Comment (kosiakkatrina) [reply app/controllers/check_errors_controller.rb:18]: So there kinda is a reason I'm doing this. I think some of our pages have additional validations checking that when there are 2 questions on the page, all of the questions have to be answered. So when we choose to clear a single question from a page with 2 questions, we get a validation error. For example, if we have uprn known - yes and a uprn value, when we choose to clear uprn to correct a composite error, the log would go into an invalid state (another example of this where the questions aren’t conditional is income frequency and earnings for lettings)

File: app/views/check_errors/confirm_clear_all_answers.html.erb (hunk 2 above)
Comment (kosiakkatrina) [reply app/views/check_errors/confirm_clear_all_answers.html.erb:10]: Like conditional questions? Yeah, maybe a warning, let me check with design

**Review by RachaelBooth — APPROVED** [review]
LGTM, subject to whatever design say about the warning

**Pull request merged**